### PR TITLE
LibWeb: Implement multiple import map support

### PR DESCRIPTION
**Pull request by shannonbooth**

### CODE REVIEW

**Comment by shannonbooth** [comment]
(WPT import doesnt work here as those tests rely on the python server thing in the harness)

File: Libraries/LibWeb/HTML/Scripting/ImportMap.cpp [hunk 1]
```diff
@@ -263,4 +266,134 @@ WebIDL::ExceptionOr<ModuleIntegrityMap> normalize_module_integrity_map(JS::Realm
...
+                    //    then:
+                    if (specifier_key.view() == record.specifier
+                        || (specifier_key.ends_with('/')
+                            && Infra::is_code_unit_prefix(specifier_key, record.specifier)
+                            && record.specifier_is_null_or_url_like_that_is_special)) {
+                        // 1. The user agent may report a warning to the console indicating the ignored rule. They
+                        //    may choose to avoid reporting if the rule is identical to an existing one.
+                        auto& console = realm.intrinsics().console_object()->console();
+                        console.output_debug_message(JS::Console::LogLevel::Warn,
+                            MUST(String::formatted("An import map rule for specifier '{}' was ignored as onet was already present in the existing import map", specifier_key)));
```
Comment (Lubrsi) [inline Libraries/LibWeb/HTML/Scripting/ImportMap.cpp:335]: nit:
```suggestion
                            MUST(String::formatted("An import map rule for specifier '{}' was ignored as one was already present in the existing import map", specifier_key)));
```

File: Libraries/LibWeb/HTML/Scripting/ImportMap.cpp [hunk 2]
```diff
@@ -263,4 +266,134 @@ WebIDL::ExceptionOr<ModuleIntegrityMap> normalize_module_integrity_map(JS::Realm
...
+    // 6. For each record of global's resolved module set:
+    for (auto const& record : global.resolved_module_set()) {
+        // 1. For each specifier → url of newImportMapImports:
+        new_import_map_imports.remove_all_matching([&](ByteString const& specifier, Optional<URL::URL> const&) {
+            // 1. If specifier starts with record's specifier, then:
+            if (specifier.starts_with(record.specifier)) {
+                // 1. The user agent may report a warning to the console indicating the ignored rule. They may choose to
+                //    avoid reporting if the rule is identical to an existing one.
+                auto& console = realm.intrinsics().console_object()->console();
+                console.output_debug_message(JS::Console::LogLevel::Warn,
```
Comment (Lubrsi) [inline Libraries/LibWeb/HTML/Scripting/ImportMap.cpp:384]: praise: I didn't know we could do this, nice alternative to plain dbgln :^)

File: Libraries/LibWeb/HTML/Window.h [hunk 3]
```diff
@@ -269,11 +288,18 @@ class Window final
...
+    // https://html.spec.whatwg.org/multipage/webappapis.html#concept-global-import-map
+    // A global object has an import map, initially an empty import map.
     ImportMap m_import_map;
 
-    // https://html.spec.whatwg.org/multipage/webappapis.html#import-maps-allowed
-    bool m_import_maps_allowed { true };
+    // https://html.spec.whatwg.org/multipage/webappapis.html#resolved-module-set
+    // A global object has a resolved module set, a set of specifier resolution records, initially empty.
+    //
+    // NOTE: The resolved module set ensures that module specifier resolution returns the same result when called
```
Comment (Lubrsi) [inline Libraries/LibWeb/HTML/Window.h:298]: nit:
```suggestion
    // Spec-Note: The resolved module set ensures that module specifier resolution returns the same result when called
```

**Review by Lubrsi — CHANGES_REQUESTED** [review]
Lokking good, just a couple nitpicks :^)

**Review by Lubrsi — APPROVED** [review]

**Pull request merged**